src/components/Calculator.jsx

```diff
@@ -0,0 +1,166 @@
+import { useState } from "react";
+import Button from "./Button";
+
+const Calculator = () => {
+  const [num1, setNum1] = useState("");
+  const [num2, setNum2] = useState("");
+  const [operator, setOperator] = useState("");
+  const [result, setResult] = useState("");
+
+  const handleNumClick = (num) => {
+    if (operator === "") {
+      setNum1(num1 + num);
+    } else {
+      setNum2(num2 + num);
+    }
+  };
+
+  const handleOperatorClick = (op) => {
+    if (op === "=") {
+      if (operator === "+") {
+        setResult(parseInt(num1) + parseInt(num2));
+      } else if (operator === "-") {
+        setResult(parseInt(num1) - parseInt(num2));
+      } else if (operator === "*") {
+        setResult(parseInt(num1) * parseInt(num2));
+      } else if (operator === "/") {
+        setResult(parseInt(num1) / parseInt(num2));
+      }
```

Comment:
I think it should be much ideal to use a switch case instead of using else-if statement, as switch is much faster and clear. but if you feel like it's okay with you to implement using else-if also it can be good as this codebase is not really big.